### PR TITLE
Sharing a files with two users

### PR DESCRIPTION
**Pull request by bishalrijal**
Sharing a files with two users acceptance test added 
Demonstrating bug for #695

### CODE REVIEW

File: tests/acceptance/features/webUIActivity/sharingInternal.feature [hunk 1]
```diff
@@ -82,3 +82,14 @@ Feature: Sharing file/folders activities
...
+  Scenario: Sharing a file with a user should be listed in the activity list of a sharer
+    Given these users have been created with default attributes but not initialized:
+      | username |
+      | user1    |
+      | user2    |
+    And user "user0" has shared file "textfile0.txt" with user "user1"
+    And user "user0" has shared file "textfile0.txt" with user "user2"
+    When user "user0" browses to the activity page
+    Then the activity number 1 should have a message saying that you have shared file "textfile0.txt" with user "User Two"
+    And the activity number 2 should have a message saying that you have shared file "textfile0.txt" with user "User One"
```
Comment (skshetry) [inline tests/acceptance/features/webUIActivity/sharingInternal.feature]: Idk if this step is a requirement. Maybe this comes as a collection (i.e.. `You have shared file "textfile0.txt" with user "User Two" and "User One"`)?

File: tests/acceptance/features/webUIActivity/sharingInternal.feature [hunk 2]
```diff
@@ -82,3 +82,14 @@ Feature: Sharing file/folders activities
     Then the activity should not have any message with keyword "shared"
     When the user filters activity list by "Shares"
     Then the activity number 1 should have a message saying that user "User Zero" has shared "simple-empty-folder, lorem.txt, folder with space and textfile0.txt" with you
+
```
Comment (skshetry) [inline tests/acceptance/features/webUIActivity/sharingInternal.feature]:
```suggestion

  @issue-695
```

Comment (skshetry) [reply tests/acceptance/features/webUIActivity/sharingInternal.feature]: We tag the bug with the respective issue.

**Comment by skshetry** [comment]
This seems to be the issue with the parser that generates `subject_prepared` because, on the activity API, it returns as a `collection` but doesn't have the former event inside it(in the DB, the event is recorded as seen in `oc_activity` table).

File: tests/acceptance/features/webUIActivity/sharingInternal.feature [hunk 3]
```diff
@@ -82,3 +82,14 @@ Feature: Sharing file/folders activities
...
+  Scenario: Sharing a file with a user should be listed in the activity list of a sharer
+    Given these users have been created with default attributes but not initialized:
+      | username |
+      | user1    |
+      | user2    |
+    And user "user0" has shared file "textfile0.txt" with user "user1"
+    And user "user0" has shared file "textfile0.txt" with user "user2"
+    When user "user0" browses to the activity page
```
Comment (skshetry) [inline tests/acceptance/features/webUIActivity/sharingInternal.feature]:
```suggestion
    When the user browses to the activity page
```

Comment (skshetry) [reply tests/acceptance/features/webUIActivity/sharingInternal.feature]: Drones didn't even run this test. 
```
--- FeatureContext has missing steps. Define them with these snippets:

    /**
     * @When user :arg1 browses to the activity page
     */
    public function userBrowsesToTheActivityPage($arg1)
    {
        throw new PendingException();
    }
```

**Comment by skshetry** [comment]
@bishalrijal, please sync the branch with the master. There were recent changes to the used steps.

File: tests/acceptance/features/webUIActivity/sharingInternal.feature [hunk 4]
```diff
@@ -86,3 +86,15 @@ Feature: Sharing file/folders activities
     Then the activity should not have any message with keyword "shared"
     When the user filters activity list by "Shares"
     Then the activity number 1 should have a message saying that user "User Zero" has shared "simple-empty-folder, lorem.txt, folder with space and textfile0.txt" with you
+
+  @issue-695
+  Scenario: Sharing a file with a user should be listed in the activity list of a sharer
+    And user "user1" has been created with default attributes and without skeleton files
```
Comment (dpakach) [inline tests/acceptance/features/webUIActivity/sharingInternal.feature]:
```suggestion
    Given user "user1" has been created with default attributes and without skeleton files
```

**Comment by individual-it** [comment]
@ownclouders rebase

**Comment by skshetry** [comment]
@individual-it @phil-davis  can you review this?

**Comment by individual-it** [comment]
sorry was forgotten

**Pull request merged**